community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -458,10 +460,20 @@ private void commit() throws TransactionFailureException
 
         try ( CommitEvent commitEvent = transactionEvent.beginCommitEvent() )
         {
-            // Trigger transaction "before" hooks
-            if ( (hooksState = hooks.beforeCommit( txState, this, storeLayer )) != null && hooksState.failed() )
+            // Trigger transaction "before" hooks.
+            if ( hasChanges() )
```

Comment:
Although what would the purpose of getting such notification be? There would be no data to try and figure out what the transaction was about. The only useful thing would be to look at the stack trace to see where the transaction came from... doesn't strike me as useful.